src/Node/SetNode.php

```diff
@@ -33,7 +33,7 @@ public function __construct(bool $capture, Node $names, Node $values, int $linen
         $safe = false;
         if ($capture) {
             $safe = true;
-            if (Node::class === get_class($values) && !count($values)) {
+            if ($values instanceof Nodes && !count($values)) {
```

Comment:
this should probably keep a `|| Node::class === get_class($values)` for BC reasons

Reply:
Indeed, added